### PR TITLE
feat: implement sequencer api with mysql

### LINKED ISSUE #489: Implement Sequencer API component: mysql
**What would you like to be added**:
Add Sequencer API component: mysql #438

### PR DESCRIPTION
**Pull request by GimmeCyy**
**What this PR does**:
add sequencer api component by mysql

**Which issue(s) this PR fixes**:
Fixes #489

### CODE REVIEW

**Comment by seeflood** [comment]
Thanks!
To fix the CI error, you can `make format` and them git commit the changes

**Comment by ZLBer** [comment]
@GimmeCyy I answered you  below https://github.com/mosn/layotto/pull/509

**Review by ZLBer — APPROVED** [review]
finally add some docs  below `docs/en/component_specs/sequencer` `docs/zh/component_specs/sequencer`

File: cmd/layotto/main.go [hunk 1]
```diff
@@ -397,6 +398,9 @@ func NewRuntimeGrpcServer(data json.RawMessage, opts ...grpc.ServerOption) (mgrp
 			runtime_sequencer.NewFactory("in-memory", func() sequencer.Store {
 				return sequencer_inmemory.NewInMemorySequencer()
 			}),
+			runtime_sequencer.NewFactory("mysql", func() sequencer.Store {
```
Comment (seeflood) [inline cmd/layotto/main.go:401]: Could u add it into the `cmd/layotto_multiple_api/main.go` too ?

**Review by seeflood — CHANGES_REQUESTED** [review]

File: docs/en/component_specs/sequencer/mysql.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# mysql
+
+## metadata fields
```
Comment (seeflood) [inline docs/en/component_specs/sequencer/mysql.md:3]: Please add this doc into the sidebar.
See https://mosn.io/layotto/#/zh/development/contributing-doc?id=_3-%e6%96%b0%e5%a2%9e%e6%96%87%e6%a1%a3%e9%9c%80%e8%a6%81%e5%81%9a%e5%93%aa%e4%ba%9b%e4%ba%8b
image

File: docs/zh/component_specs/sequencer/mysql.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# mysql
+
+## 配置项说明
```
Comment (seeflood) [inline docs/zh/component_specs/sequencer/mysql.md:3]: Please add this doc into the sidebar.
See https://mosn.io/layotto/#/zh/development/contributing-doc?id=_3-%e6%96%b0%e5%a2%9e%e6%96%87%e6%a1%a3%e9%9c%80%e8%a6%81%e5%81%9a%e5%93%aa%e4%ba%9b%e4%ba%8b
image

**Review by seeflood — CHANGES_REQUESTED** [review]

File: components/sequencer/mysql/mysql.go [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+	begin, err := metadata.Db.Begin()
+	if err != nil {
+		return nil, err
+	}
+	err = begin.QueryRow("SELECT sequencer_key, sequencer_value, version FROM ? WHERE sequencer_key = ?", metadata.TableName, req.Key).Scan(&Key, &Value, &oldVersion)
+
+	if err == sql.ErrNoRows {
+		Value = 1
+		Version = 1
+		_, err := begin.Exec("INSERT INTO ?(sequencer_key, sequencer_value, version) VALUES(?,?,?)", metadata.TableName, req.Key, Value, Version)
```
Comment (ZLBer) [inline components/sequencer/mysql/mysql.go:97]: 这里会有并发插入问题吗？

**Comment by ZLBer** [comment]
@GimmeCyy 本地的锁似乎是没用的，layotto可能是多机部署。可以先交流下再写代码~

**Review by ZLBer — APPROVED** [review]
LGTM

**Review by seeflood — APPROVED** [review]
lgtm.

**Review by zhenjunMa — APPROVED** [review]

**Comment by seeflood** [comment]
@GimmeCyy Thanks for your help!

**Pull request merged**